Source/WebCore/html/canvas/GPUCanvasContext.cpp

```diff
@@ -116,21 +116,21 @@ void GPUCanvasContext::createSwapChainIfNeeded()
 
     GPUSurfaceDescriptor surfaceDescriptor = {
         { "WebGPU Canvas surface"_s },
-        GPUExtent3DDict { static_cast<uint32_t>(m_width), static_cast<uint32_t>(m_height), 1 },
-        1 /* sampleCount */,
-        m_configuration->format,
-        m_configuration->usage
+        // FIXME: Include the CompositorIntegration here.
     };
 
     m_surface = m_configuration->device->createSurface(surfaceDescriptor);
     ASSERT(m_surface);
 
     GPUSwapChainDescriptor descriptor = {
         { "WebGPU Canvas swap chain"_s },
-        GPUExtent3DDict { static_cast<uint32_t>(m_width), static_cast<uint32_t>(m_height), 1 },
-        1 /* sampleCount */,
         m_configuration->format,
-        m_configuration->usage
+        m_configuration->usage,
+        m_configuration->viewFormats,
+        m_configuration->colorSpace,
+        m_configuration->compositingAlphaMode,
+        static_cast<uint32_t>(m_width), // FIXME: Is it possible for these to be negative?
```

Comment:
Isn't it not recommended in C++ to use unsigned types simply because the value is only expected to be positive?

The C++ core guidelines recommend against using [unsigned to avoid negative values](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-nonnegative), Section 4.4 on page 73 of The C++ Programming Language says unsigned types should be used for bitfields and not in an attempt to ensure values are positive, other talks by people on the C++ standards committee (e.g., Herb Sutter) recommend against using unsigned types simply because the value is expected to by positive. Plus I think it was one of the reasons for std::ssize in C++20

Do we have official guidance in WebKit? If not, I think it would make more sense to preferred signed types for values which are not bitfields.

Reply:
Oh, this is the first I've heard about this recommendation. I'm just using unsigneds here because that's what WebGPU.h uses: https://github.com/webgpu-native/webgpu-headers/blob/main/webgpu.h#L975

So the cast is going to happen at _some_ point, either here or in `DeviceImpl::createSwapChain()`. Do you have a preference where the cast happens?

(I also disagree with those C++ guidelines, but that's a discussion for another time...)

Reply:
I don’t have any strong opinions since it’s unlikely we’ll perform arithmetic or loops on these integers, it would be fine for me to be uint32_t in this case

Reply:
🆗